packages/beacon-node/src/chain/prepareNextSlot.ts

```diff
@@ -240,4 +244,26 @@ export class PrepareNextSlotScheduler {
     state.hashTreeRoot();
     hashTreeRootTimer?.();
   }
+
+  /**
+   * Stop eth1 data polling after eth1DepositIndex has reached depositRequestsStartIndex in Electra as described in EIP-6110
+   */
+  stopEth1Polling(): void {
+    // Only continue if eth1 is still polling. State regen is expensive
+    if (this.chain.eth1.isPollingEth1Data()) {
+      const finalizedCheckpoint = this.chain.forkChoice.getFinalizedCheckpoint();
+      const fork = this.config.getForkInfoAtEpoch(finalizedCheckpoint.epoch).name;
+
+      if (isForkPostElectra(fork)) {
```

Comment:
can't we just check the fork based on current slot? it's good to only stop polling if finalized state is electra but I think we shouldn't even call `getStateByCheckpoint` if it's not even electra fork

Reply:
> can't we just check the fork based on current slot?

Just checking slot based on clock is not enough. There is a short transition period in Electra where we still need to poll and process eth1 deposits. We can only stop polling when all the eth1 deposits have been processed

> it's good to only stop polling if finalized state is electra but I think we shouldn't even call getStateByCheckpoint if it's not even electra fork

I see. Added fork based on clock slot as pre-condition

Reply:
> There is a short transition period in Electra where we still need to poll and process eth1 deposits

I know but at the earliest we would run the deposit index check if it's post-electra based on clock slot, and then we can load the state and check the index